runtime/decimal.c

```diff
@@ -57,6 +57,20 @@ TaggedPtrPanicCode _bal_decimal_div(TaggedPtr tp1, TaggedPtr tp2) {
     return finish(&d, &cx); 
 }
 
+TaggedPtrPanicCode _bal_decimal_rem(TaggedPtr tp1, TaggedPtr tp2) {
+    decQuad d;
+    decContext cx;
+    initContext(&cx);
+    decQuadRemainder(&d, taggedToDecQuad(tp1), taggedToDecQuad(tp2), &cx);
+    return finish(&d, &cx);    
+}
+
+TaggedPtrPanicCode _bal_decimal_neg(TaggedPtr tp) {
+    decQuad d;
+    decQuadCopyNegate(&d, taggedToDecQuad(tp));
+    return finish(&d, NULL);
```

Comment:
I don't like this passing in NULL to finish. Will this work?

```
const decQuad *dp = taggedToDecQuad(tp);
if (decQuadClass(dp) == DEC_CLASS_POS_ZERO) {
    return tp;
}
decQuad d;
decQuadCopyNegate(&d, taggedToDecQuad(tp));
return createDecimal(&d);
```

Reply:
This is logically correct, but the representation of the result fails at this point 0.0. Since we return the tp at zero(0.0), we are getting the 0.0 but in the subset we should get only 0.
I am going to do this like,
```
TaggedPtr _bal_decimal_neg(TaggedPtr tp) {
    const decQuad *dq = taggedToDecQuad(tp);
    decQuad d;
    if (decQuadClass(dq) == DEC_CLASS_POS_ZERO) {
        decQuadZero(&d);
    } 
    else {
        decQuadCopyNegate(&d, dq);
    }
    return createDecimal(&d);
}
```